eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/SubmittedVariant.java

```diff
@@ -232,71 +243,28 @@ public void setValidated(Boolean validated) {
 
     @Override
     public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (!(o instanceof SubmittedVariant)) {
-            return false;
-        }
-
-        SubmittedVariant variant = (SubmittedVariant) o;
-
-        if (taxonomyAccession != variant.taxonomyAccession) {
-            return false;
-        }
-        if (start != variant.start) {
-            return false;
-        }
-        if (!referenceSequenceAccession.equals(variant.referenceSequenceAccession)) {
-            return false;
-        }
-        if (!projectAccession.equals(variant.projectAccession)) {
-            return false;
-        }
-        if (!contig.equals(variant.contig)) {
-            return false;
-        }
-        if (!referenceAllele.equals(variant.referenceAllele)) {
-            return false;
-        }
-        if (!alternateAllele.equals(variant.alternateAllele)) {
-            return false;
-        }
-        if (clusteredVariantAccession != null ? !clusteredVariantAccession.equals(
-                variant.clusteredVariantAccession) : variant.clusteredVariantAccession != null) {
-            return false;
-        }
-        if (supportedByEvidence != null ? !supportedByEvidence.equals(
-                variant.supportedByEvidence) : variant.supportedByEvidence != null) {
-            return false;
-        }
-        if (assemblyMatch != null ? !assemblyMatch.equals(variant.assemblyMatch) : variant.assemblyMatch != null) {
-            return false;
-        }
-        if (allelesMatch != null ? !allelesMatch.equals(variant.allelesMatch) : variant.allelesMatch != null) {
-            return false;
-        }
-        if (validated != null ? !validated.equals(variant.validated) : variant.validated != null) {
-            return false;
-        }
-        return true;
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        SubmittedVariant that = (SubmittedVariant) o;
+        return taxonomyAccession == that.taxonomyAccession &&
+                start == that.start &&
+                Objects.equals(referenceSequenceAccession, that.referenceSequenceAccession) &&
+                Objects.equals(projectAccession, that.projectAccession) &&
+                Objects.equals(contig, that.contig) &&
+                Objects.equals(referenceAllele, that.referenceAllele) &&
+                Objects.equals(alternateAllele, that.alternateAllele) &&
+                Objects.equals(clusteredVariantAccession, that.clusteredVariantAccession) &&
+                Objects.equals(supportedByEvidence, that.supportedByEvidence) &&
+                Objects.equals(assemblyMatch, that.assemblyMatch) &&
+                Objects.equals(allelesMatch, that.allelesMatch) &&
+                Objects.equals(validated, that.validated);
```

Comment:
Have you verified the equivalence of these statements against the previous (more verbose) code? I am hoping the complex ternary conditionals for attributes like supportedByEvidence in the previous code are covered in this terse validation..

Reply:
Yes, they are equivalent